src/auth/index.ts

```diff
@@ -161,22 +202,45 @@ export class Authenticator {
 
   async authenticate(): Promise<void> {
     // Try to get token from cache using read lock
-    const cachedToken = await this.tryGetCachedToken();
+    const cachedToken = await this.tryGetCachedTokenRecord();
     if (cachedToken) {
-      this.accessToken = cachedToken;
+      this.accessToken = cachedToken.token;
+      this.tokenExpiryTimestampMs = cachedToken.tokenExpiryTimestampMs;
       return;
     }
-
     // No cached token, acquire write lock and authenticate
     await this.acquireWriteLockAndAuthenticate();
   }
 
-  private async tryGetCachedToken(): Promise<string | undefined> {
+  async reAuthenticate(): Promise<void> {
```

Comment:
Can this function just be replaced by adding a `skipCache` parameter to `acquireWriteLockAndAuthenticate`? Or just use it under the hood? Since currently a lot of logic seems to be duplicated in these two

Reply:
It does a different thing, `acquireWriteLockAndAuthenticate` checks cache then performs auth, while this function clears cache under write lock and does auth. We can add a flag that would mean either cache is cleared or fetched, but I think it will hurt the readability - it will be another layer of ifs.

Reply:
Taking into account my previous confusion, WDYT about removing the cache check from `acquireWriteLockAndAuthenticate` to leave it a "single responsibility", and just use it in other function which will take care of the cache?